mlflow/projects/utils.py

```diff
@@ -50,16 +50,21 @@
 def _parse_subdirectory(uri):
     # Parses a uri and returns the uri and subdirectory as separate values.
     # Uses '#' as a delimiter.
+    unquoted_uri = _remove_quotes(uri)
     subdirectory = ""
-    parsed_uri = uri
-    if "#" in uri:
-        subdirectory = uri[uri.find("#") + 1 :]
-        parsed_uri = uri[: uri.find("#")]
+    parsed_uri = unquoted_uri
+    if "#" in unquoted_uri:
+        subdirectory = unquoted_uri[unquoted_uri.find("#") + 1 :]
+        parsed_uri = unquoted_uri[: unquoted_uri.find("#")]
     if subdirectory and "." in subdirectory:
         raise ExecutionException("'.' is not allowed in project subdirectory paths.")
     return parsed_uri, subdirectory
 
 
+def _remove_quotes(uri):
```

Comment:
```suggestion
def _strip_quotes(uri):
```

Can we rename this function?

Reply:
Thanks for the feedback! function `_remove_quotes` renamed to `_strip_quotes`. I've already done another force push.